core/src/test/java/org/apache/iceberg/TestFastAppend.java

```diff
@@ -277,4 +278,27 @@ public void testRecoveryWithoutManifestList() {
     Assert.assertTrue("Should commit the same new manifest",
         metadata.currentSnapshot().manifests().contains(newManifest));
   }
+
+  @Test
+  public void testAppendFilesFromTable() {
+    table.newFastAppend()
+        .appendFile(FILE_A)
+        .appendFile(FILE_B)
+        .commit();
+
+    // collect data files from deserialization
+    Iterator<DataFile> filesIterator = table.currentSnapshot().addedFiles().iterator();
+
+    table.newDelete().deleteFile(FILE_A).deleteFile(FILE_B).commit();
+
+    Snapshot oldSnapshot = table.currentSnapshot();
+
+    AppendFiles fastAppend = table.newFastAppend();
+    while (filesIterator.hasNext()) {
```

Comment:
Why not use `for (DataFile file : table.currentSnapshot().addedFiles())`? Using a while loop and an explicit iterator seems odd.

Reply:
Make sense.